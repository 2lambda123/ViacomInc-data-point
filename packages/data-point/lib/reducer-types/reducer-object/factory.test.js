/* eslint-env jest */

const factory = require('./factory')
const createReducer = require('../index').create
const constant = require('../..').helpers.constant

test('ReducerObject.factory#isType', () => {
  expect(factory.isType('$foo')).toBe(false)
  expect(factory.isType(() => true)).toBe(false)
  expect(factory.isType([])).toBe(false)
  expect(factory.isType({})).toBe(true)
})

describe('ReducerObject.factory#getProps', () => {
  it('should accept an empty object', () => {
    const reducer = factory.getProps(createReducer, {})
    expect(reducer).toEqual({
      constants: {},
      reducers: []
    })
  })
  it('should accept a non-empty object', () => {
    const props = factory.getProps(createReducer, {
      a1: '$a[]',
      b1: {
        a2: ['$a2', () => false],
        b2: [
          {
            a3: 'transform:entity-name',
            b3: '$b1.b2.b3',
            c3: constant('CONSTANT')
          }
        ],
        c2: constant(1)
      },
      c1: constant({
        a: 1,
        b: 2
      }),
      d1: constant(5)
    })
    expect(props).toMatchSnapshot()
  })
})

describe('ReducerObject.factory#create', () => {
  it('reducer object with no props argument', () => {
    const reducer = factory.create(createReducer)
    expect(reducer).toMatchSnapshot()
    expect(reducer.source()).toEqual({})
  })
  it('reducer object with props argument', () => {
    const reducer = factory.create(createReducer, {
      a: '$a',
      b: '$b',
      c: constant('c')
    })
    expect(reducer).toMatchSnapshot()
    expect(reducer.source()).toEqual({ c: 'c' })
  })
})
