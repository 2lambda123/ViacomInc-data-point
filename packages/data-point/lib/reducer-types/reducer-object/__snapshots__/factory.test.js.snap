// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer/reducer-function#getReducerProps should accept a non-empty object 1`] = `
Array [
  Object {
    "path": Array [
      "a1",
    ],
    "reducer": ReducerPath {
      "asCollection": true,
      "body": [Function],
      "name": "a",
      "type": "ReducerPath",
    },
  },
  Object {
    "path": Array [
      "b1",
      "a2",
    ],
    "reducer": ReducerList {
      "reducers": Array [
        ReducerPath {
          "asCollection": false,
          "body": [Function],
          "name": "a2",
          "type": "ReducerPath",
        },
        ReducerFunction {
          "body": [Function],
          "type": "ReducerFunction",
        },
      ],
      "type": "ReducerList",
    },
  },
  Object {
    "path": Array [
      "b1",
      "b2",
    ],
    "reducer": ReducerObject {
      "props": Array [
        Object {
          "path": Array [
            "a3",
          ],
          "reducer": ReducerEntity {
            "asCollection": false,
            "entityType": "transform",
            "hasEmptyConditional": false,
            "id": "transform:entity-name",
            "name": "entity-name",
            "type": "ReducerEntity",
            "withEmptyConditional": false,
          },
        },
        Object {
          "path": Array [
            "b3",
          ],
          "reducer": ReducerPath {
            "asCollection": false,
            "body": [Function],
            "name": "b1.b2.b3",
            "type": "ReducerPath",
          },
        },
      ],
      "type": "ReducerObject",
    },
  },
]
`;
